binary_search_tree/tree.py

```diff
@@ -17,41 +17,129 @@ def __init__(self):
     # Time Complexity: 
     # Space Complexity: 
     def add(self, key, value = None):
-        pass
+        if self.root == None:
+            self.root = TreeNode(key,value)
+        current = self.root
+
+        while True:
```

Comment:
Just a personal note.  I dislike `while True` loops which depend on `break` statements to control the flow of execution because it makes the loop harder to trace and understand.